raven/src/main/java/org/opencadc/raven/FilesAction.java

```diff
@@ -0,0 +1,140 @@
+*  the GNU Affero General Public        la “GNU Affero General Public
+*  License as published by the          License” telle que publiée
+*  Free Software Foundation,            par la Free Software Foundation
+*  either version 3 of the              : soit la version 3 de cette
+*  License, or (at your option)         licence, soit (à votre gré)
+*  any later version.                   toute version ultérieure.
+*
+*  OpenCADC is distributed in the       OpenCADC est distribué
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+************************************************************************
+*/
+
+package org.opencadc.raven;
+
+import ca.nrc.cadc.rest.InlineContentHandler;
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.InventoryUtil;
+
+/**
+ * Base class for handling "files" requests
+ *
+ * @author adriand
+ */
+public abstract class FilesAction extends ArtifactAction {
+
+    private static final Logger log = Logger.getLogger(FilesAction.class);
+
+    /**
+     * Default, no-arg constructor.
+     */
+    public FilesAction() {
+        super();
+    }
+
+    // constructor for unit tests with no config/init
+    FilesAction(boolean init) {
+        super(init);
+    }
+
+    @Override
+    /**
+     * Parse the request path.
+     */
+    void parseRequest() throws Exception {
+        parsePath(syncInput.getPath());
+    }
+
+    void parsePath(final String path) {
+        log.debug("path: " + path);
+        if (path == null) {
+            throw new IllegalArgumentException("path expected");
+        }
+        String au = path;
+        if (path.indexOf(":") < 0) {
+            au = "cadc:" + path;
+        }
+        artifactURI = createArtifactURI(au);
+    }
+
+    /**
+     * Create a valid artifact uri.
+     * @param uri The input string.
+     * @return The artifact uri object.
+     */
+    private URI createArtifactURI(String uri) {
+        try {
+            log.debug("artifactURI: " + uri);
+            artifactURI = new URI(uri);
```

Comment:
This is assigning to the member var and returning it for another assignment (after checking). Use a local variable here instead of assignment side effect.

Reply:
Doh! Fixed